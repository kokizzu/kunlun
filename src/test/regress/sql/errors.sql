--
-- ERRORS
--

-- bad in postquel, but ok in PostgreSQL
select 1;


--
-- UNSUPPORTED STUFF

-- doesn't work
-- notify pg_class
--

--
-- SELECT

-- this used to be a syntax error, but now we allow an empty target list
select;

-- no such relation
select * from nonesuch;

-- bad name in target list
select nonesuch from pg_database;

-- empty distinct list isn't OK
select distinct from pg_database;

-- bad attribute name on lhs of operator
select * from pg_database where nonesuch = pg_database.datname;

-- bad attribute name on rhs of operator
select * from pg_database where pg_database.datname = nonesuch;

-- bad attribute name in select distinct on
select distinct on (foobar) * from pg_database;


--
-- DELETE

-- missing relation name (this had better not wildcard!)
delete from;

-- no such relation
delete from nonesuch;


--
-- DROP

-- missing relation name (this had better not wildcard!)
--DDL_STATEMENT_BEGIN--
drop table;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
-- no such relation
drop table nonesuch;
--DDL_STATEMENT_END--


--
-- ALTER TABLE

-- relation renaming

-- missing relation name
--DDL_STATEMENT_BEGIN--
alter table rename;
--DDL_STATEMENT_END--

-- no such relation
--DDL_STATEMENT_BEGIN--
alter table nonesuch rename to newnonesuch;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such relation
alter table nonesuch rename to stud_emp;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- conflict
alter table stud_emp rename to aggtest;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- self-conflict
alter table stud_emp rename to stud_emp;
--DDL_STATEMENT_END--


-- attribute renaming

--DDL_STATEMENT_BEGIN--
-- no such relation
alter table nonesuchrel rename column nonesuchatt to newnonesuchatt;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such attribute
alter table emp rename column nonesuchatt to newnonesuchatt;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- conflict
alter table emp rename column salary to manager;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- conflict
alter table emp rename column salary to oid;
--DDL_STATEMENT_END--


--
-- TRANSACTION STUFF

-- not in a xact
abort;

-- not in a xact
end;


--
-- CREATE AGGREGATE

-- sfunc/finalfunc type disagreement
--DDL_STATEMENT_BEGIN--
create aggregate newavg2 (sfunc = int4pl,
			  basetype = int4,
			  stype = int4,
			  finalfunc = int2um,
			  initcond = '0');
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- left out basetype
create aggregate newcnt1 (sfunc = int4inc,
			  stype = int4,
			  initcond = '0');
			  

--DDL_STATEMENT_END--
--
-- DROP INDEX

--DDL_STATEMENT_BEGIN--
-- missing index name
drop index;

--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
-- bad index name
drop index 314159;

--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
-- no such index
drop index nonesuch;
--DDL_STATEMENT_END--


--
-- DROP AGGREGATE

--DDL_STATEMENT_BEGIN--

-- missing aggregate name
drop aggregate;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- missing aggregate type
drop aggregate newcnt1;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad aggregate name
drop aggregate 314159 (int);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad aggregate type
drop aggregate newcnt (nonesuch);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such aggregate
drop aggregate nonesuch (int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such aggregate for type
drop aggregate newcnt (float4);

--DDL_STATEMENT_END--

--
-- DROP FUNCTION

--DDL_STATEMENT_BEGIN--
-- missing function name
drop function ();
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad function name
drop function 314159();
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such function
drop function nonesuch();
--DDL_STATEMENT_END--


--
-- DROP TYPE

--DDL_STATEMENT_BEGIN--
-- missing type name
drop type;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad type name
drop type 314159;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type
drop type nonesuch;


--DDL_STATEMENT_END--
--
-- DROP OPERATOR

--DDL_STATEMENT_BEGIN--
-- missing everything
drop operator;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad operator name
drop operator equals;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- missing type list
drop operator ===;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- missing parentheses
drop operator int4, int4;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- missing operator name
drop operator (int4, int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- missing type list contents
drop operator === ();
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such operator
drop operator === (int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such operator by that name
drop operator === (int4, int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type1
drop operator = (nonesuch);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type1
drop operator = ( , int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type1
drop operator = (nonesuch, int4);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type2
drop operator = (int4, nonesuch);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such type2
drop operator = (int4, );
--DDL_STATEMENT_END--

--
-- DROP RULE

--DDL_STATEMENT_BEGIN--
-- missing rule name
drop rule;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- bad rule name
drop rule 314159;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- no such rule
drop rule nonesuch on noplace;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- these postquel variants are no longer supported
drop tuple rule nonesuch;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
drop instance rule nonesuch on noplace;
--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
drop rewrite rule nonesuch;
--DDL_STATEMENT_END--

--
-- Check that division-by-zero is properly caught.
--

select 1/0;

select 1::int8/0;

select 1/0::int8;

select 1::int2/0;

select 1/0::int2;

select 1::numeric/0;

select 1/0::numeric;

select 1::float8/0;

select 1/0::float8;

select 1::float4/0;

select 1/0::float4;


--
-- Test psql's reporting of syntax error location
--

xxx;

--DDL_STATEMENT_BEGIN--
CREATE foo;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE TABLE ;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
CREATE TABLE
--DDL_STATEMENT_END--
\g

INSERT INTO foo VALUES(123) foo;

INSERT INTO 123
VALUES(123);

INSERT INTO foo
VALUES(123) 123
;

-- with a tab
--DDL_STATEMENT_BEGIN--
CREATE TABLE foo
  (id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY,
	id3 INTEGER NOT NUL,
   id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- long line to be truncated on the left
CREATE TABLE foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- long line to be truncated on the right
CREATE TABLE foo(
id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL, id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY);

--DDL_STATEMENT_END--
--DDL_STATEMENT_BEGIN--
-- long line to be truncated both ways
CREATE TABLE foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
--DDL_STATEMENT_END--

-- long line to be truncated on the left, many lines
--DDL_STATEMENT_BEGIN--
CREATE
TEMPORARY
TABLE
foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
id4 INT4
UNIQUE
NOT
NULL,
id5 varchar(50)
UNIQUE
NOT
NULL)
;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- long line to be truncated on the right, many lines
CREATE
TEMPORARY
TABLE
foo(
id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL, id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY)
;
--DDL_STATEMENT_END--

--DDL_STATEMENT_BEGIN--
-- long line to be truncated both ways, many lines
CREATE
TEMPORARY
TABLE
foo
(id
INT4
UNIQUE NOT NULL, idx INT4 UNIQUE NOT NULL, idy INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL,
idz INT4 UNIQUE NOT NULL,
idv INT4 UNIQUE NOT NULL);
--DDL_STATEMENT_END--

-- more than 10 lines...
--DDL_STATEMENT_BEGIN--
CREATE
TEMPORARY
TABLE
foo
(id
INT4
UNIQUE
NOT
NULL
,
idm
INT4
UNIQUE
NOT
NULL,
idx INT4 UNIQUE NOT NULL, idy INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL,
idz INT4 UNIQUE NOT NULL,
idv
INT4
UNIQUE
NOT
NULL);
--DDL_STATEMENT_END--

-- Check that stack depth detection mechanism works and
-- max_stack_depth is not set too high
--DDL_STATEMENT_BEGIN--
create function infinite_recurse() returns int as
'select infinite_recurse()' language sql;
--DDL_STATEMENT_END--
\set VERBOSITY terse
select infinite_recurse();
