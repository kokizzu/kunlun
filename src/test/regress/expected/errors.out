--
-- ERRORS
--
-- bad in postquel, but ok in PostgreSQL
select 1;
 ?column? 
----------
        1
(1 row)

--
-- UNSUPPORTED STUFF
-- doesn't work
-- notify pg_class
--
--
-- SELECT
-- this used to be a syntax error, but now we allow an empty target list
select;
--
(1 row)

-- no such relation
select * from nonesuch;
psql:sql/errors.sql:23: ERROR:  relation "nonesuch" does not exist
LINE 1: select * from nonesuch;
                      ^
-- bad name in target list
select nonesuch from pg_database;
psql:sql/errors.sql:26: ERROR:  column "nonesuch" does not exist
LINE 1: select nonesuch from pg_database;
               ^
-- empty distinct list isn't OK
select distinct from pg_database;
psql:sql/errors.sql:29: ERROR:  syntax error at or near "from"
LINE 1: select distinct from pg_database;
                        ^
-- bad attribute name on lhs of operator
select * from pg_database where nonesuch = pg_database.datname;
psql:sql/errors.sql:32: ERROR:  column "nonesuch" does not exist
LINE 1: select * from pg_database where nonesuch = pg_database.datna...
                                        ^
-- bad attribute name on rhs of operator
select * from pg_database where pg_database.datname = nonesuch;
psql:sql/errors.sql:35: ERROR:  column "nonesuch" does not exist
LINE 1: ...ect * from pg_database where pg_database.datname = nonesuch;
                                                              ^
-- bad attribute name in select distinct on
select distinct on (foobar) * from pg_database;
psql:sql/errors.sql:38: ERROR:  column "foobar" does not exist
LINE 1: select distinct on (foobar) * from pg_database;
                            ^
--
-- DELETE
-- missing relation name (this had better not wildcard!)
delete from;
psql:sql/errors.sql:45: ERROR:  syntax error at or near ";"
LINE 1: delete from;
                   ^
-- no such relation
delete from nonesuch;
psql:sql/errors.sql:48: ERROR:  relation "nonesuch" does not exist
LINE 1: delete from nonesuch;
                    ^
--
-- DROP
-- missing relation name (this had better not wildcard!)
drop table;
psql:sql/errors.sql:55: ERROR:  syntax error at or near ";"
LINE 1: drop table;
                  ^
-- no such relation
drop table nonesuch;
psql:sql/errors.sql:57: ERROR:  table "nonesuch" does not exist
--
-- ALTER TABLE
-- relation renaming
-- missing relation name
alter table rename;
psql:sql/errors.sql:66: ERROR:  syntax error at or near ";"
LINE 1: alter table rename;
                          ^
-- no such relation
alter table nonesuch rename to newnonesuch;
psql:sql/errors.sql:69: ERROR:  relation "nonesuch" does not exist
-- no such relation
alter table nonesuch rename to stud_emp;
psql:sql/errors.sql:72: ERROR:  relation "nonesuch" does not exist
-- conflict
alter table stud_emp rename to aggtest;
psql:sql/errors.sql:75: ERROR:  relation "aggtest" already exists
-- self-conflict
alter table stud_emp rename to stud_emp;
psql:sql/errors.sql:78: ERROR:  relation "stud_emp" already exists
-- attribute renaming
-- no such relation
alter table nonesuchrel rename column nonesuchatt to newnonesuchatt;
psql:sql/errors.sql:84: ERROR:  relation "nonesuchrel" does not exist
-- no such attribute
alter table emp rename column nonesuchatt to newnonesuchatt;
psql:sql/errors.sql:87: ERROR:  column "nonesuchatt" does not exist
-- conflict
alter table emp rename column salary to manager;
psql:sql/errors.sql:90: ERROR:  column "manager" of relation "stud_emp" already exists
-- conflict
alter table emp rename column salary to oid;
ALTER TABLE
--
-- TRANSACTION STUFF
-- not in a xact
abort;
psql:sql/errors.sql:100: WARNING:  there is no transaction in progress
ROLLBACK
-- not in a xact
end;
psql:sql/errors.sql:103: WARNING:  there is no transaction in progress
COMMIT
--
-- CREATE AGGREGATE
-- sfunc/finalfunc type disagreement
create aggregate newavg2 (sfunc = int4pl,
			  basetype = int4,
			  stype = int4,
			  finalfunc = int2um,
			  initcond = '0');
psql:sql/errors.sql:114: ERROR:  function int2um(smallint) requires run-time type coercion
-- left out basetype
create aggregate newcnt1 (sfunc = int4inc,
			  stype = int4,
			  initcond = '0');
psql:sql/errors.sql:119: ERROR:  aggregate input type must be specified
			  
--
-- DROP INDEX
-- missing index name
drop index;
psql:sql/errors.sql:126: ERROR:  syntax error at or near ";"
LINE 1: drop index;
                  ^
-- bad index name
drop index 314159;
psql:sql/errors.sql:129: ERROR:  syntax error at or near "314159"
LINE 1: drop index 314159;
                   ^
-- no such index
drop index nonesuch;
psql:sql/errors.sql:132: ERROR:  index "nonesuch" does not exist
--
-- DROP AGGREGATE
-- missing aggregate name
drop aggregate;
psql:sql/errors.sql:140: ERROR:  syntax error at or near ";"
LINE 1: drop aggregate;
                      ^
-- missing aggregate type
drop aggregate newcnt1;
psql:sql/errors.sql:143: ERROR:  syntax error at or near ";"
LINE 1: drop aggregate newcnt1;
                              ^
-- bad aggregate name
drop aggregate 314159 (int);
psql:sql/errors.sql:146: ERROR:  syntax error at or near "314159"
LINE 1: drop aggregate 314159 (int);
                       ^
-- bad aggregate type
drop aggregate newcnt (nonesuch);
psql:sql/errors.sql:149: ERROR:  type "nonesuch" does not exist
-- no such aggregate
drop aggregate nonesuch (int4);
psql:sql/errors.sql:152: ERROR:  aggregate nonesuch(integer) does not exist
-- no such aggregate for type
drop aggregate newcnt (float4);
psql:sql/errors.sql:155: ERROR:  aggregate newcnt(real) does not exist
--
-- DROP FUNCTION
-- missing function name
drop function ();
psql:sql/errors.sql:162: ERROR:  syntax error at or near "("
LINE 1: drop function ();
                      ^
-- bad function name
drop function 314159();
psql:sql/errors.sql:165: ERROR:  syntax error at or near "314159"
LINE 1: drop function 314159();
                      ^
-- no such function
drop function nonesuch();
psql:sql/errors.sql:168: ERROR:  function nonesuch() does not exist
--
-- DROP TYPE
-- missing type name
drop type;
psql:sql/errors.sql:175: ERROR:  syntax error at or near ";"
LINE 1: drop type;
                 ^
-- bad type name
drop type 314159;
psql:sql/errors.sql:178: ERROR:  syntax error at or near "314159"
LINE 1: drop type 314159;
                  ^
-- no such type
drop type nonesuch;
psql:sql/errors.sql:181: ERROR:  type "nonesuch" does not exist
--
-- DROP OPERATOR
-- missing everything
drop operator;
psql:sql/errors.sql:188: ERROR:  syntax error at or near ";"
LINE 1: drop operator;
                     ^
-- bad operator name
drop operator equals;
psql:sql/errors.sql:191: ERROR:  syntax error at or near ";"
LINE 1: drop operator equals;
                            ^
-- missing type list
drop operator ===;
psql:sql/errors.sql:194: ERROR:  syntax error at or near ";"
LINE 1: drop operator ===;
                         ^
-- missing parentheses
drop operator int4, int4;
psql:sql/errors.sql:197: ERROR:  syntax error at or near ","
LINE 1: drop operator int4, int4;
                          ^
-- missing operator name
drop operator (int4, int4);
psql:sql/errors.sql:200: ERROR:  syntax error at or near "("
LINE 1: drop operator (int4, int4);
                      ^
-- missing type list contents
drop operator === ();
psql:sql/errors.sql:203: ERROR:  syntax error at or near ")"
LINE 1: drop operator === ();
                           ^
-- no such operator
drop operator === (int4);
psql:sql/errors.sql:206: ERROR:  missing argument
LINE 1: drop operator === (int4);
                               ^
HINT:  Use NONE to denote the missing argument of a unary operator.
-- no such operator by that name
drop operator === (int4, int4);
psql:sql/errors.sql:209: ERROR:  operator does not exist: integer === integer
-- no such type1
drop operator = (nonesuch);
psql:sql/errors.sql:212: ERROR:  missing argument
LINE 1: drop operator = (nonesuch);
                                 ^
HINT:  Use NONE to denote the missing argument of a unary operator.
-- no such type1
drop operator = ( , int4);
psql:sql/errors.sql:215: ERROR:  syntax error at or near ","
LINE 1: drop operator = ( , int4);
                          ^
-- no such type1
drop operator = (nonesuch, int4);
psql:sql/errors.sql:218: ERROR:  type "nonesuch" does not exist
-- no such type2
drop operator = (int4, nonesuch);
psql:sql/errors.sql:221: ERROR:  type "nonesuch" does not exist
-- no such type2
drop operator = (int4, );
psql:sql/errors.sql:224: ERROR:  syntax error at or near ")"
LINE 1: drop operator = (int4, );
                               ^
--
-- DROP RULE
-- missing rule name
drop rule;
psql:sql/errors.sql:230: ERROR:  syntax error at or near ";"
LINE 1: drop rule;
                 ^
-- bad rule name
drop rule 314159;
psql:sql/errors.sql:233: ERROR:  syntax error at or near "314159"
LINE 1: drop rule 314159;
                  ^
-- no such rule
drop rule nonesuch on noplace;
psql:sql/errors.sql:236: ERROR:  relation "noplace" does not exist
-- these postquel variants are no longer supported
drop tuple rule nonesuch;
psql:sql/errors.sql:239: ERROR:  syntax error at or near "tuple"
LINE 1: drop tuple rule nonesuch;
             ^
drop instance rule nonesuch on noplace;
psql:sql/errors.sql:240: ERROR:  syntax error at or near "instance"
LINE 1: drop instance rule nonesuch on noplace;
             ^
drop rewrite rule nonesuch;
psql:sql/errors.sql:241: ERROR:  syntax error at or near "rewrite"
LINE 1: drop rewrite rule nonesuch;
             ^
--
-- Check that division-by-zero is properly caught.
--
select 1/0;
psql:sql/errors.sql:247: ERROR:  division by zero
select 1::int8/0;
psql:sql/errors.sql:249: ERROR:  division by zero
select 1/0::int8;
psql:sql/errors.sql:251: ERROR:  division by zero
select 1::int2/0;
psql:sql/errors.sql:253: ERROR:  division by zero
select 1/0::int2;
psql:sql/errors.sql:255: ERROR:  division by zero
select 1::numeric/0;
psql:sql/errors.sql:257: ERROR:  division by zero
select 1/0::numeric;
psql:sql/errors.sql:259: ERROR:  division by zero
select 1::float8/0;
psql:sql/errors.sql:261: ERROR:  division by zero
select 1/0::float8;
psql:sql/errors.sql:263: ERROR:  division by zero
select 1::float4/0;
psql:sql/errors.sql:265: ERROR:  division by zero
select 1/0::float4;
psql:sql/errors.sql:267: ERROR:  division by zero
--
-- Test psql's reporting of syntax error location
--
xxx;
psql:sql/errors.sql:274: ERROR:  syntax error at or near "xxx"
LINE 1: xxx;
        ^
CREATE foo;
psql:sql/errors.sql:276: ERROR:  syntax error at or near "foo"
LINE 1: CREATE foo;
               ^
CREATE TABLE ;
psql:sql/errors.sql:278: ERROR:  syntax error at or near ";"
LINE 1: CREATE TABLE ;
                     ^
CREATE TABLE
\g
psql:sql/errors.sql:281: ERROR:  syntax error at end of input
LINE 1: CREATE TABLE
                    ^
INSERT INTO foo VALUES(123) foo;
psql:sql/errors.sql:283: ERROR:  syntax error at or near "foo"
LINE 1: INSERT INTO foo VALUES(123) foo;
                                    ^
INSERT INTO 123
VALUES(123);
psql:sql/errors.sql:286: ERROR:  syntax error at or near "123"
LINE 1: INSERT INTO 123
                    ^
INSERT INTO foo
VALUES(123) 123
;
psql:sql/errors.sql:290: ERROR:  syntax error at or near "123"
LINE 2: VALUES(123) 123
                    ^
-- with a tab
CREATE TABLE foo
  (id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY,
	id3 INTEGER NOT NUL,
   id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
psql:sql/errors.sql:296: ERROR:  syntax error at or near "NUL"
LINE 3:  id3 INTEGER NOT NUL,
                         ^
-- long line to be truncated on the left
CREATE TABLE foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
psql:sql/errors.sql:300: ERROR:  syntax error at or near "NUL"
LINE 1: ..., id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
                                                                   ^
-- long line to be truncated on the right
CREATE TABLE foo(
id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL, id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY);
psql:sql/errors.sql:304: ERROR:  syntax error at or near "NUL"
LINE 2: id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(5...
                        ^
-- long line to be truncated both ways
CREATE TABLE foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL);
psql:sql/errors.sql:307: ERROR:  syntax error at or near "NUL"
LINE 1: ...varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 I...
                                                             ^
-- long line to be truncated on the left, many lines
CREATE
TEMPORARY
TABLE
foo(id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
id4 INT4
UNIQUE
NOT
NULL,
id5 varchar(50)
UNIQUE
NOT
NULL)
;
psql:sql/errors.sql:322: ERROR:  syntax error at or near "NUL"
LINE 4: ..., id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL,
                                                                   ^
-- long line to be truncated on the right, many lines
CREATE
TEMPORARY
TABLE
foo(
id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL, id INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY)
;
psql:sql/errors.sql:330: ERROR:  syntax error at or near "NUL"
LINE 5: id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(5...
                        ^
-- long line to be truncated both ways, many lines
CREATE
TEMPORARY
TABLE
foo
(id
INT4
UNIQUE NOT NULL, idx INT4 UNIQUE NOT NULL, idy INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL,
idz INT4 UNIQUE NOT NULL,
idv INT4 UNIQUE NOT NULL);
psql:sql/errors.sql:341: ERROR:  syntax error at or near "NUL"
LINE 7: ...varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 I...
                                                             ^
-- more than 10 lines...
CREATE
TEMPORARY
TABLE
foo
(id
INT4
UNIQUE
NOT
NULL
,
idm
INT4
UNIQUE
NOT
NULL,
idx INT4 UNIQUE NOT NULL, idy INT4 UNIQUE NOT NULL, id2 varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 INT4 UNIQUE NOT NULL, id5 varchar(50) UNIQUE NOT NULL,
idz INT4 UNIQUE NOT NULL,
idv
INT4
UNIQUE
NOT
NULL);
psql:sql/errors.sql:365: ERROR:  syntax error at or near "NUL"
LINE 16: ...varchar(50) NOT NULL PRIMARY KEY, id3 INTEGER NOT NUL, id4 I...
                                                              ^
-- Check that stack depth detection mechanism works and
-- max_stack_depth is not set too high
create function infinite_recurse() returns int as
'select infinite_recurse()' language sql;
CREATE FUNCTION
\set VERBOSITY terse
select infinite_recurse();
psql:sql/errors.sql:372: ERROR:  stack depth limit exceeded
